File: typescript/behavioral/visitor/guru_example/concreteComponentA.ts
import {Component} from "./component";
import {Visitor} from "./visitor";

export class ConcreteComponentA implements Component {
  public accept(visitor: Visitor) {
    visitor.visitConcreteComponentA(this)
  }

  public exclusiveMethodOfConcreteComponentA(): String {
    return 'ConcreteComponentA: exclusiveMethodOfConcreteComponentA.'
  }
}

File: dart/behavioral/visitor/guru_example/bin/concrete_component_a.dart
import 'component.dart';
import 'visitor.dart';

class ConcreteComponentA implements Component {
  @override
  void accept(Visitor visitor) {
    visitor.visitConcreteComponentA(this);
  }

  String a() {
    return 'ConcreteComponentA: a.';
  }
}
